bblocks/import_tools/debt/wb_ids.py

```diff
@@ -0,0 +1,213 @@
+        ).astype({"year": "datetime64[ns]"})
+
+        # Save the data to a feather file
+        data.to_feather(self._path / f"{indicator}_{start_year}-{end_year}.feather")
+
+        return self
+
+    def load_data(
+        self, indicators: str | list, start_year: int, end_year: int
+    ) -> ImportData:
+        """Load the data for an indicator or a list of indicators.
+
+        Args:
+            indicators: The indicator(s) to load. They must be in the IDS format
+                (e.g. DT.DOD.DECT.CD). To view all available indicators, call
+                `.get_available_indicators()`.
+            start_year: The first year to include in the data
+            end_year: The last year to include in the data
+
+        """
+
+        # check if indicators is a string and convert to a list.
+        if isinstance(indicators, str):
+            indicators = [indicators]
+
+        # check if the indicators are valid
+        for indicator in indicators:
+            if indicator not in self.get_available_indicators().keys():
+                raise ValueError(f"{indicator} is not a valid indicator.")
+
+            # check if the data is already stored locally
+            stored_data = self._check_stored_data(
+                indicator, start_year=start_year, end_year=end_year
+            )
+
+            # if the data isn't stored locally, get it
+            if not stored_data:
+                # get the data
+                self._get_indicator(indicator, start_year, end_year)
+
+                # Once the data has been downloaded, save the filename to the
+                # stored_data variable.
+                stored_data = f"{indicator}_{start_year}-{end_year}.feather"
+
+            # load the data into the data dictionary, keeping only requested years.
+            self._data[f"{indicator}_{start_year}-{end_year}"] = (
+                pd.read_feather(self._path / stored_data)
+                .query(f"year >= {start_year} and year <= {end_year}")
+                .reset_index(drop=True)
+            )
+
+        return self
+
+    def update_data(self, reload_data: bool = True) -> ImportData:
+        """Update the data for all loaded indicators."""
+
+        # check if any data has been loaded
+        if len(self._data) < 1:
+            raise ValueError("No data has been loaded.")
+
+        # Update the data for each indicator
+        for indicator in self._data:
+            # get the indicator name, start year and end year
+            ind, start, end = self._indicator_parameters(indicator)
+            # get the data
+            self._get_indicator(indicator=ind, start_year=start, end_year=end)
+
+            # reload the data if reload_data is True
+            if reload_data:
+                self.load_data(indicators=ind, start_year=start, end_year=end)
+
+        return self
+
+    def get_data(self, indicators: str | list = "all", **kwargs) -> pd.DataFrame:
```

Comment:
There is issue in `get_data` that has to do with how the data is stored. Since data for a single indicator is stored with a start and end date. if the same indicator is loaded twice for different dates, some of which overlap, this causes some duplicated to be present in the dataframe

Reply:
Good catch. Now it will check that the indicator hasn't been loaded previously. If it has, it will raise a KeyError saying that the same indicator can only be loaded once at a time.